Protorev query highest liquidity pools

NotJeremyLiu opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
v    If your PR doesn't close an issue, that's OK!  Just remove the Closes: #XXX line!
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## What is the purpose of the change

- Add visibility into protorev for the highest liquidity pool stored given a denom pair via a query endpoint

## Brief Changelog

- Adds rpc / cli queries that return a pool id given a denom pair to look up in protorev's highest liquidity pool method

## Testing and Verifying

- All tests still pass
- Added test to verify new query
- Verified CLI works via localosmosis

## Documentation and Release Note

- Added new queries to documentation

NotJeremyLiu commented:
Added label for non-state breaking, don't know if it should be backport or no-backport

davidterpay commented on proto/osmosis/protorev/v1beta1/query.proto:
```diff
@@ -108,6 +108,12 @@ service Query {
     option (google.api.http).get = "/osmosis/v14/protorev/enabled";
   }
+
+  // GetProtoRevPool queries whether the module is enabled or not
```
Do we want this to also return the hot routes as well or only the highest liquidity pool?
```suggestion
  // GetProtoRevPool queries the pool id used for arbitrage route building given a pair of denominations
```

davidterpay replied:
If we only care about the highest liquidity pool id, I would update the doc to say that.

davidterpay commented on x/protorev/client/cli/query.go:
```diff
@@ -148,6 +149,14 @@ func NewQueryPoolWeightsCmd() (*osmocli.QueryDescriptor, *types.QueryGetProtoRev
+func NewQueryPoolCmd() (*osmocli.QueryDescriptor, *types.QueryGetProtoRevPoolRequest) {
+	return &osmocli.QueryDescriptor{
+		Use:   "pool [base_denom] [other_denom]",
+		Short: "Query the pool id for a given denom pair stored in ProtoRev",
```
Update this based on the comment above.

NotJeremyLiu replied:
yeah, just want highest liquidity pools. Specified here: https://github.com/osmosis-labs/osmosis/pull/4829/commits/821b247669016401caec222de0743718f194e4e8

davidterpay reviewed (commented):
LGTM with small nits.

p0mvn commented on x/protorev/keeper/grpc_query_test.go:
```diff
@@ -355,3 +355,25 @@ func (suite *KeeperTestSuite) TestGetProtoRevEnabledQuery() {
 	suite.Require().Equal(enabled, res.Enabled)
 }
+
+// TestGetProtoRevStatisticsByRoute tests the query for statistics by route
```
```suggestion
// TestGetProtoRevPool tests the query for statistics by route
```

NotJeremyLiu replied:
fixed here: https://github.com/osmosis-labs/osmosis/pull/4829/commits/2dbb66767fd88b55d3713610d7cbb0953f0d859b

p0mvn reviewed (approved):
LGTM

p0mvn commented:
Let's add a changelog entry for this and merge

NotJeremyLiu commented:
> Let's add a changelog entry for this and merge

Added in: https://github.com/osmosis-labs/osmosis/pull/4829/commits/7049f018ac75d82185bddc1819d0722e041c51a0 and rebased on main, so should be ready to merge.
